extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,343 @@
+					isReadOnly: true,
+					width: '30%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				}
+			],
+			data: [this.coordinatorData.map(p => [p.condition, p.details, p.lastUpdate])]
+		}).component();
+
+		this.podDropDown = this.modelView.modelBuilder.dropDown().withProps({ width: '150px' }).component();
+		this.disposables.push(
+			this.podDropDown.onValueChanged(() => {
+				this.podConditionsTable.setFilter(this.podConditionsTableIndexes.get(String(this.podDropDown.value)));
+			})
+		);
+
+		this.podConditionsContainer.addItem(this.podDropDown, { CSSStyles: { 'margin': '10px 0px 10px 0px' } });
+		this.podConditionsContainer.addItem(this.podConditionsTable);
+		this.podConditionsLoading = this.modelView.modelBuilder.loadingComponent()
+			.withItem(this.podConditionsContainer)
+			.withProperties<azdata.LoadingComponentProperties>({
+				loading: !this._postgresModel.configLastUpdated
+			}).component();
+
+		this.refreshPodCondtions();
+
+		content.addItem(this.podConditionsLoading, { CSSStyles: cssStyles.text });
+
+		this.initialized = true;
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// Refresh
+		const refreshButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: loc.refresh,
+			iconPath: IconPathHelper.refresh
+		}).component();
+
+		this.disposables.push(
+			refreshButton.onDidClick(async () => {
+				refreshButton.enabled = false;
+				try {
+					this.podConditionsLoading!.loading = true;
+
+					await Promise.all([
+						this._postgresModel.refresh(),
+						this._controllerModel.refresh()
+					]);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+				finally {
+					refreshButton.enabled = true;
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: refreshButton, toolbarSeparatorAfter: true }
+		]).component();
+	}
+
+	private createPodList(): string[] {
+		const podStatus = this._postgresModel.config?.status.podsStatus;
+		let podNames: string[] = [];
+
+		podStatus?.forEach(p => {
+			let podHealthModels: PodHealthModel[] = [];
+			let indexes: number[] = [];
+
+
+			p.conditions.forEach(c => {
+				const condtionContainer = this.modelView.modelBuilder.flexContainer().withProps({
+					CSSStyles: { 'alignItems': 'center', 'height': '15px' }
+				}).component();
+
+				if (c.status === 'False') {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.fail,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+						value: c.message,
+					}).component());
+				} else {
```

Comment:
I think if this section did the component creation/adding one time, it would reduce duplication across branches.  Something like:

```
let icon;
let message;

if (false) {
    icon = fail
    message = c.message
} else {
    icon = success
    if (Initialized) {
        message = loc.podInitialized
    } else if { ... }
}

condtionContainer.addItem(this.modelView.modelBuilder.image(icon...
condtionContainer.addItem(this.modelView.modelBuilder.text(message...
```